File: src/gestorAplicacion/Hija/ContactosLocales.java
package gestorAplicacion.Hija;
import gestorAplicacion.Padre.*;

import java.util.ArrayList;


//Los contactos locales son personas con las que ya se inicio una convesacion
class ContactosLocales extends Contacto{
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	private ArrayList<Mensaje> mensajes;

	//Genera los contactos locales con la capacidad de recibir mensajes
	ContactosLocales(String nombre, String email, String nombreCompleto) {
		super(nombre, email, nombreCompleto);
		mensajes = new ArrayList<Mensaje>();
	}

	ArrayList<Mensaje> getMensajes() {
		return mensajes;
	}

	void setMensajes(ArrayList<Mensaje> mensajes) {
		this.mensajes = mensajes;
	}

	@Override
	public String toString() {
		return super.toString();
	}

	@Override
	protected String getNombre() {
		return super.getNombre();
	}

	@Override
	protected void setNombre(String nombre) {
		super.setNombre(nombre);
	}

	@Override
	protected String getEmail() {
		return super.getEmail();
	}

	@Override
	protected void setEmail(String email) {
		super.setEmail(email);
	}

	@Override
	protected String getNombreCompleto() {
		return super.getNombreCompleto();
	}

	@Override
	protected void setNombreCompleto(String nombreCompleto) {
		super.setNombreCompleto(nombreCompleto);
	}
	
}

File: Python/uiMain/gestorAplicacion/Hija/contactosLocales.py
from gestorAplicacion.Padre.Contacto import Contacto
from gestorAplicacion.Padre.Mensaje import Mensaje
#Los contactos locales son personas con las que ya se inicio una convesacion
class ContactosLocales(Contacto):

    _mensajes: list[Mensaje]
    #Genera los contactos locales con la capacidad de recibir mensajes
    def __init__(self, nombre, email, nombreCompleto):
        super().__init__(nombre, email, nombreCompleto)
        self._mensajes = []

    def getMensajes(self) -> list[Mensaje]:
        return self._mensajes

    def setMensajes(self, mensajes):
        self._mensajes = mensajes

    def __str__(self):
        return super.__str__()

    def getNombre(self):
        return super.getNombre()
    
    def setNombre(self, nombre):
        super.setNombre(nombre)

    def getEmail(self):
        return super.getEmail()
    
    def setEmail(self, email):
        super.setEmail(email)
    
    def getNombreCompleto(self):
        return super.getNombreCompleto()
    
    def setNombreCompleto(self, nombreCompleto):
        super.setNombreCompleto(nombreCompleto)